frontend/src/component/Home/HomeNoWorkspace.jsx

```diff
@@ -16,32 +25,36 @@ export const HomeNoWorkspace = props =>
     <div className='homepagecard__delimiter delimiter primaryColorBg' />
 
     <div className='homepagecard__text'>
-      {(props.canCreateWorkspace
-        ? props.t('You will create your first space')
-        : (
-          <div className='homepagecard__text__user'>
-            <div className='homepagecard__text__user__top'>
-              {props.t("You aren't member of any space yet")}
-            </div>
-            <div>{props.t('Please refer to an administrator or a trusted user')}</div>
-          </div>
-        )
-      )}
+      <div className='homepagecard__text__user__top'>
+        {props.t("You aren't member of any space yet")}
+      </div>
+      {getWorkspaceMessage(props.canCreateWorkspace, props.canJoinWorkspace, props.t)}
```

Comment:
I'm not completely sure that you can pass props.t in parameter because of the way translation strings are parsed when building the translation list. Maybe you'll need to pass props, or use `const {props} = this` in `getWorkspaceMessage` (and yes, unfortunately make it depend on `this`.

(@Skylsmoi ?)

Reply:
It will work because build-translation script looks for `props.t()` and `t()` functions. I don't think that this is a great pattern though.

Alternatively, you can keep your old way of calling props.t on the function return and add a fake function `const f = p => p` for `getWorkspaceMessage` to trick the build-translation script into adding the key. But that might be an even worst pattern

If I remember correctly, there is a way to `require` or `import` the translation function from i18n. The way we use in Tracim is using the HOC to get it but that isn't the only way. It might be cleaner that passing it through a parameter. But then, your function `getWorkspaceMessage` wont be pure anymore. I don't know what is the best way. The one that is currently commited is fine to me